### PR TITLE
Geomancer abilities, merits, and adjustments

### PR DESCRIPTION
**Pull request by ibm2431**
I needed to take Omni's geo branch into our repository to regain control over pushing commits to it.

---

This is the work relating to Geomancer abilities, merits, and other adjustments made since the initial April submission of #494.

I have not reviewed it yet.

As @Omnione won't be the author of this Pull Request - and therefore won't be able to push additional commits - review items will have to be addressed in a future Pull Request.

Please review as if this were a normal Pull Request. We'll compile a list of work items that needs to be done for this branch prior to a merge into canary.

### CODE REVIEW

**Comment by TeoTwawki** [comment]
> I needed to take Omni's geo branch into our repository to regain control over pushing commits to it.

~~There is a setting that allows upstreams (thats topaz) to push to forks on the fork persons end (omni) just fyi~~ Beta feature, disregard. The existing feature everyone has is to allow per-pullrequest, and per repository isn't a thing (yet). An alternative is to add collaborators though, which would work that way (full repo access).

**Comment by ibm2431** [comment]
There is, and it's enabled by default when a downstream opens a PR, but apparently it doesn't enable if someone other than the downstream owner opens the PR (the case in #700).

File: scripts/globals/magic.lua [hunk 1]
```diff
@@ -5,12 +5,33 @@ require("scripts/globals/weather")
...
+    [8]  = {direction = '[SSE]', MATT = 0.25, MACC = 0.75, MAG_BURST_BONUS = 0.00, MAGIC_CRITHITRATE = 0.00},
+    [9]  = {direction = '[S]',   MATT = 0.00, MACC = 1.00, MAG_BURST_BONUS = 0.00, MAGIC_CRITHITRATE = 0.00},
+    [10] = {direction = '[SSW]', MATT = 0.00, MACC = 0.75, MAG_BURST_BONUS = 0.25, MAGIC_CRITHITRATE = 0.00},
+    [11] = {direction = '[SW]',  MATT = 0.00, MACC = 0.50, MAG_BURST_BONUS = 0.50, MAGIC_CRITHITRATE = 0.00},
+    [12] = {direction = '[WSW]', MATT = 0.00, MACC = 0.25, MAG_BURST_BONUS = 0.75, MAGIC_CRITHITRATE = 0.00},
+    [13] = {direction = '[W]',   MATT = 0.00, MACC = 0.00, MAG_BURST_BONUS = 1.00, MAGIC_CRITHITRATE = 0.00},
+    [14] = {direction = '[WNW]', MATT = 0.00, MACC = 0.00, MAG_BURST_BONUS = 0.75, MAGIC_CRITHITRATE = 0.25},
+    [15] = {direction = '[NW]',  MATT = 0.00, MACC = 0.00, MAG_BURST_BONUS = 0.50, MAGIC_CRITHITRATE = 0.50},
+    [16] = {direction = '[NNW]', MATT = 0.00, MACC = 0.00, MAG_BURST_BONUS = 0.25, MAGIC_CRITHITRATE = 0.75},
+}
```
Comment (cocosolos) [inline scripts/globals/magic.lua:33]: Not a fan of this, can't we do some math to calculate these bonuses?

File: src/map/lua/lua_baseentity.cpp [hunk 2]
```diff
@@ -2452,6 +2453,47 @@ inline int32 CLuaBaseEntity::getAngle(lua_State *L)
...
+    else if (cardinalAngle <= 152 && cardinalAngle > 136){ cardinalQuadrant = 7; } // SE
+    else if (cardinalAngle <= 136 && cardinalAngle > 120){ cardinalQuadrant = 8; } // SSE
+    else if (cardinalAngle <= 120 && cardinalAngle > 104){ cardinalQuadrant = 9; } // S
+    else if (cardinalAngle <= 104 && cardinalAngle > 88){ cardinalQuadrant = 10;}  // SSW
+    else if (cardinalAngle <= 88 && cardinalAngle > 72){ cardinalQuadrant = 11; }  // SW
+    else if (cardinalAngle <= 72 && cardinalAngle > 56){ cardinalQuadrant = 12; }  // WSW
+    else if (cardinalAngle <= 56 && cardinalAngle > 40){ cardinalQuadrant = 13; }  // W
+    else if (cardinalAngle <= 40 && cardinalAngle > 24){ cardinalQuadrant = 14; }  // WNW
+    else if (cardinalAngle <= 24 && cardinalAngle > 8){ cardinalQuadrant = 15; }   // NW
+    else if (cardinalAngle <= 8 && cardinalAngle > 0){ cardinalQuadrant = 16; }    // NNW
```
Comment (cocosolos) [inline src/map/lua/lua_baseentity.cpp:2492]: There has to be a better way to do this.

Comment (Omnione) [reply src/map/lua/lua_baseentity.cpp:2492]: Yeah im sure someone smarter than me could think of a way, i tried, but it never seemed right

File: scripts/globals/magic.lua (hunk 1 above)
Comment (Omnione) [reply scripts/globals/magic.lua:33]: i think if the direction stuff can be made better with claculations, this could be improved also

**Comment by ibm2431** [comment]
Applying a hold until after geomancer bubbles~

**Pull request merged**